ui/app/components/account-menu/index.js

```diff
@@ -103,6 +119,53 @@ AccountMenu.prototype.render = function () {
   ])
 }
 
+
+/**
+ * Renders the Account-New Menu-Entries dynamically, using keyringProviders data
+ * 
+ * Relevant data is passed within the kyeringProviders state object
+ * 
+ */
+AccountMenu.prototype.renderAccountNew = function () {
+
+  // UI type will be later supplemented with e.g. "HardWalletPage"
+
+
+  const {
+    keyrings,
+    keyringProviders,
+    showNewAccountPage, // needed thus onClick event fires properly
+  } = this.props
+
+  var temp=[];
+  Object.keys(keyringProviders).map((key, index) => {
+
+    const func = keyringProviders[key].func;
+    const text = keyringProviders[key].text;
+    const img  = keyringProviders[key].img;
```

Comment:
You could do this in one line using destructuring (very fancy!):
`const { func, text, img } = keyringProviders[key]`